components/eamxx/cime_config/SystemTests/ksxx.py

```diff
@@ -0,0 +1,646 @@
+        help="Name of the variable set to analyze.",
+    )
+
+    parser.add_argument(
+        "--ninst",
+        default=30,
+        type=int,
+        help="The number of instances (should be the same for " "both cases).",
+    )
+
+    parser.add_argument(
+        "--critical",
+        default=13,
+        type=int,
+        help=(
+            "The critical value (desired significance level) for rejecting the "
+            "null hypothesis. Only used when --uncorrected / -u flag is passed. "
+            "Otherwise the critical value is 1."
+        ),
+    )
+
+    parser.add_argument("--img-dir", default=os.getcwd(), help="Image output location.")
+
+    parser.add_argument(
+        "--img-fmt", default="png", type=str, help="Format for output images"
+    )
+
+    parser.add_argument(
+        "--component", default="eam", help="Model component name (e.g. eam, cam, ...)"
+    )
+
+    parser.add_argument(
+        "--uncorrected",
+        "-u",
+        action="store_true",
+        default=False,
+        help="Do not use FDR correction to compute global pass / fail, will ",
+    )
+
+    parser.add_argument(
+        "--alpha", default=0.05, type=float, help="Alpha threshold for pass / fail"
+    )
+
+    args, _ = parser.parse_known_args(args)
+
+    # use config file arguments, but override with command line arguments
+    if args.config:
+        default_args = parser.parse_args([])
+
+        for (
+            key,
+            val,
+        ) in vars(args).items():
+            if val != vars(default_args)[key]:
+                args.config["ks"][key] = val
+
+        config_arg_list = []
+        _ = [
+            config_arg_list.extend(["--" + key, str(val)])
+            for key, val in args.config["ks"].items()
+            if key != "config"
+        ]
+        args, _ = parser.parse_known_args(config_arg_list)
+
+    return args
+
+
+def col_fmt(dat):
+    """Format results for table output."""
+    if dat is None:
+        _out = "-"
+    elif isinstance(dat, (tuple, list)):
+        try:
+            _out = "{:.3e}, {:.3e}".format(*dat)
+        except TypeError:
+            _out = dat
+    elif isinstance(dat, float):
+        _out = f"{dat:.3e}"
+    else:
+        _out = "-"
+
+    return _out
+
+
+def run(name, config):
+    """
+    Runs the analysis.
+
+    Args:
+        name: The name of the test
+        config: A dictionary representation of the configuration file
+
+    Returns:
+       The result of elements.page with the list of elements to display
+    """
+
+    config_arg_list = []
+    [config_arg_list.extend(["--" + key, str(val)]) for key, val in config.items()]
+
+    args = parse_args(config_arg_list)
+
+    args.img_dir = os.path.join(livvkit.output_dir, "validation", "imgs", name)
+    fn.mkdir_p(args.img_dir)
+
+    details, img_gal = main(args)
+
+    table_data = pd.DataFrame(details).T
+    uc_rejections = (table_data["K-S test p-val"] < args.alpha).sum()
+    _hdrs = [
+        "h",
```

Comment:
Maybe pedantic, but the "h0" here originally refers to null hypothesis acceptance/rejection rather than h0 from the hist files (only changes the text of output tables on the generated webpage)